src/Http/Http.Results/src/HttpResultsHelper.cs

```diff
@@ -15,6 +17,9 @@ internal static partial class HttpResultsHelper
     internal const string DefaultContentType = "text/plain; charset=utf-8";
     private static readonly Encoding DefaultEncoding = Encoding.UTF8;
 
+    // Remove once https://github.com/dotnet/aspnetcore/pull/45886 is done.
```

Comment:
Hmm, there are a mix of safe and unsafe overloads here. @brunolins16 I think I'll need a hand from you figuring out the right way to make this safe.

Reply:
What I have as suggestions is this: https://github.com/dotnet/aspnetcore/pull/46008

Basically:
1. Always use JsonTypeInfo in the helper 
2. Mark the Json methods with Options unsafe because we'll set the reflection-based provider 
3. Add more safe overloads to [Typed] Results to allow usage of JsonTypeInfo or JsonSerializerContext

Reply:
Wait isn't this safe now?

Reply:
Changes to HttpResultsHelper are required. Bruno is making the changes here: https://github.com/dotnet/aspnetcore/pull/46008. I've updated the comment to point to this PR.

Will leave suppressions here and they can be removed with Bruno's PR.